In [2]:
pip install statsmodels


[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.2.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
import numpy as np

# HYPOTHESIS TESTING

#### Hypothesis testing is a fundamental statistical method used to make inferences about population parameters based on sample data. It involves formulating two competing hypotheses: the null hypothesis (H0), which typically states that there is no effect or no difference between groups, and the alternative hypothesis (H1), which asserts the presence of an effect or difference. The process generally entails selecting an appropriate statistical test based on the nature of the data and the research question, calculating a test statistic from the sample data, and determining the probability of obtaining such results under the assumption that the null hypothesis is true, known as the p-value. If the p-value is below a predetermined significance level (usually 0.05), the null hypothesis is rejected in favor of the alternative hypothesis, suggesting evidence for the presence of an effect or difference. Otherwise, if the p-value is above the significance level, the null hypothesis is not rejected. Hypothesis testing provides a systematic framework for drawing conclusions from data, aiding researchers in making informed decisions and drawing valid inferences about the population.

## Case1

#### Q1.Suppose a child psychologist claims that the average time working mothers spend talking to their children is at least 11 minutes per day. You conduct a random sample of 1000 working mothers and find they spend an average of 11.5 minutes per day talking with their children. Assume prior research suggests the population standard deviation is 2.3 minutes.Conduct a test with a level of significance of alpha = 0.05.


## Null Hypothesis

#### Ho=μ≤11 minutes

## Alternative Hypothesis

#### Ha=μ>11 minutes

#### x̄=11.5minutes

#### n=1000

#### σ=2.3 minutes

#### α=0.05

### Statistical Test

In [ ]:

In this case we have a sampleof 1000 which is greater than 30 and here sample mean,standard deviation is already given,
therefore we are going to compare the sample mean to a population value so we are using Z-test .

    rejection criteria is z_statistics>critical value,fail to accept Ho


### Formula for Z test=(x̄ – μ) / (σ / √n)

x̄=sample mean=11.5 minutes μ=population mean=11 minutes σ=population standard deviation=2.3 minutes n=sample size=1000



In [11]:
import math

In [12]:
 Z=(11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955

### Determine Critical Value or P-Value:

Since we are conducting a one tailed test(testing if the average time is more than 11 minutes),we will look up the critical value fora z-score corresponding to an alpha level of 0.05 from z-distirbution table.


##### taking alpha level as  0.05 in this case then the critical value for the one-tailed test is 1.645

### Conclusion for the hypothesis test
Here in this case the calculated z-score is 6.88 which is much greater than the critical value 1.645 that means we have to reject the null hypothesis.Thus we concluded that the average time working mothers spend talking to their children is indeed more than 11 minutes per day.

# Case 2

#### Q2. A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.

## Null Hypothesis

#### Ho=μ≥5 minutes

## Alternative Hypothesis

#### Ha=μ<5 minutes

#### x̄=4.6 minutes

#### n=40

#### σ=0.8 minutes

#### μ=5 minutes

#### α=0.05

#### Degree of freedom(n-1)=39

### Statistical Test



#### Here the population std is unknown therefore we are using T-test for this.

If the sample size is small(less than 30) and the population standard deviation is unknown then T-test is applicable.Even for large sample size(greater than 30) the T-test can be used even if the population standard deviation is unknown this is due to Central Limit Theorem which states that the distribution of sample means approaches a normal distribution as the sample size increases.

### Formula for T-test=(x̄-μ)/(σ/√n)
x̄=Sample mean=4.6 minutes
μ=Population mean=5 minutes
σ=Sample Standard deviation=0.8 minutes


In [16]:
T_test_static=(4.6-5)/(0.8/np.sqrt(40))
print('T_statistic:',T_test_static)

T_statistic: -3.162277660168382


#### Here the function t.ppf(α,df) calculates a critical value from the T-distribution based on given significance level "α" and degrees of freedom "df".

#### The term "percent pont function"ppf gives a value from the distribution in a way that if you randomly select a value from the distribution there will be a probability that it will be less than or equal to that value.

In [17]:
#Finding critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value=t.ppf(0.05,39)
print("Critical value:",critical_value)

Critical value: -1.684875121711225


### Conclusion for the hypothesis test¶

Here in this case the calculated T-value is (-3.16) which is less than critical value (-1.685) from T distribution.Therefore we fail to accept the null hypothesis.
Thus we came into a conclusion that there is no relevant evidence that says the average waiting time for customers is less than 5 minutes 